core/dbt/parser/generic_test_builders.py

```diff
@@ -473,7 +515,9 @@ def build_raw_code(self) -> str:
 
     def build_model_str(self):
         targ = self.target
-        if isinstance(self.target, UnparsedNodeUpdate):
+        if isinstance(self.target, UnparsedModelUpdate) and self.version:
```

Comment:
What happens here if the target is an UnparsedModelUpdate without self.version?

Reply:
it's super subtle actually - but it would get handled by the `elif isinstance(self.target, UnparsedNodeUpdate)` case below because `UnparsedModelUpdate` subclasses `UnparsedNodeUpdate`. Handling that case more explicitly would probably be less error prone if there is a change to the inheritance structure between these two going forward. What do you think?

Reply:
Yeah, I think I'd prefer that case to be handled explicitly, for the reason you give. Also more readable. Though a comment would probably be okay at this point.